test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetPackageUpdateTests.cs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using FluentAssertions;
+using NuGet.CommandLine.Xplat.Tests;
+using NuGet.ProjectModel;
+using NuGet.Test.Utility;
+using NuGet.Versioning;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Dotnet.Integration.Test
+{
+    [Collection(DotnetIntegrationCollection.Name)]
+    public class DotnetPackageUpdateTests : IClassFixture<PackageSearchRunnerFixture>
+    {
+        private readonly DotnetIntegrationTestFixture _testFixture;
+        private readonly ITestOutputHelper _testOutputHelper;
+        private readonly string _xplatCli;
+        private readonly IReadOnlyDictionary<string, string> _envVars;
+
+        // The .NET SDK downloads reference assembly packages for target frameworks it can't find ref assemblies for
+        // locally. Ideally this should be solved, so it's not necessary to download the packages every time the test
+        // runs. At a minimum, this should be changed to use package source mapping, once update package supports PSM.
+        private const string NugetConfigFormat = """
+            <configuration>
+              <packageSources>
+                <clear />
+                <add key="test" value="{0}" />
+                <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+              </packageSources>
+              <packageSourceMapping>
```

Comment:
Nice! BTW, I've been wanting to add this or similar to the dotnet nuget.config templates.